src/xcode/ENA/ENA/Source/Services/HealthCertificateValidationOnboardedCountries/HealthCertificateValidationOnboardedCountriesProvider.swift

```diff
@@ -0,0 +1,166 @@
+			completion: { result in
+				switch result {
+				case let .success(countries):
+					Log.info("Successfully decoded country codes: \(private: countries). Returning now.", log: .vaccination)
+					// Save in success case for caching
+					let receivedOnboardedCountries = HealthCertificateValidationOnboardedCountriesCache(
+						onboardedCountries: countries,
+						lastOnboardedCountriesETag: eTag
+					)
+					store.validationOnboardedCountriesCache = receivedOnboardedCountries
+					completion(.success(countries))
+				case let .failure(error):
+					Log.error("Could not decode CBOR from package with error:", log: .vaccination, error: error)
+					completion(.failure(.ONBOARDED_COUNTRIES_VALIDATION_ERROR(error)))
+				}
+			}
+		)
+	}
+	
+	private func processOnboardedCountriesFailure(
+		error: URLSession.Response.Failure,
+		completion: @escaping (Result<[Country], HealthCertificateValidationOnboardedCountriesError>) -> Void
+	) {
+		switch error {
+		case .notModified:
+			// Normally we should have cached something before
+			Log.info("Download new onboarded countries aborted due to not modified content. Taking cached countries.", log: .vaccination)
+			if let cachedOnboardedCountries = store.validationOnboardedCountriesCache?.onboardedCountries {
+				completion(.success(cachedOnboardedCountries))
+			} else {
+				// If not, return edge case error
+				Log.error("Could not find cached countries but need some.", log: .vaccination)
+				completion(.failure(.ONBOARDED_COUNTRIES_MISSING_CACHE))
+			}
+		case .noNetworkConnection:
+			Log.error("Could not download onboarded countries due to no network.", log: .vaccination, error: error)
+			completion(.failure(.ONBOARDED_COUNTRIES_NO_NETWORK))
+		case let .serverError(statusCode):
+			switch statusCode {
+			case 400...409:
+				Log.error("Could not download onboarded countries due to client error with status code: \(statusCode).", log: .vaccination, error: error)
+				completion(.failure(.ONBOARDED_COUNTRIES_CLIENT_ERROR))
+			default:
+				Log.error("Could not download onboarded countries due to server error with status code: \(statusCode).", log: .vaccination, error: error)
+				completion(.failure(.ONBOARDED_COUNTRIES_SERVER_ERROR))
+			}
+		default:
+			Log.error("Could not download onboarded countries due to server error.", log: .vaccination, error: error)
+			completion(.failure(.ONBOARDED_COUNTRIES_SERVER_ERROR))
+		}
+	}
+	
+	/// Extracts by the HealthCertificateToolkit the list of countrys. Expects the list as CBOR-Data and return for success the list of Country-Objects.
+	private func extractCountryCodes(
+		cborData: Data,
+		completion: (Result<[Country], RuleValidationError>
+		) -> Void
+	) {
+		let extractOnboardedCountryCodesResult = OnboardedCountriesAccess().extractCountryCodes(from: cborData)
+		
+		switch extractOnboardedCountryCodesResult {
+		case let .success(countryCodes):
+			let countries = countryCodes.compactMap {
+				Country(countryCode: $0)
```

Comment:
I know, I made the suggestion to use `Country` here, but now, if I think again, it might be problematic. Because the `Country` implementation changes something on the country code, which is needed to get a proper localized description for the UI. But if this changed country code will be used again to validate rules, it will be wrong.

Reply:
Yeah I see the problem you mention. But when i think about it, what will happen when we pass the `[String]` (aka the real codes) to the UI? At this point, we will do the same at display something which does not exist and pass it to the validation service back?
Or should we implement a new mapping of the country codes to localized readable codes ignoring our existing `Country` implementation?

Reply:
I think we can have a typealias on String. Something like `ValidationCountryCode`. And we return [ValidationCountryCode]. And accept ValidationCountryCode for validation. To make it more clear, we want the same type and not Country.localizedName. 

Still we need to check the UI-Code to pass the right String. I will do that at the PR for the Validation-Screen.

Reply:
Sounds like a play. I will do so.

Reply:
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-8381